content/en/docs/setup/independent/kubelet-integration.md

```diff
@@ -8,113 +8,160 @@ weight: 40
+initializing. A kubelet configuration file is also written to `/etc/kubernetes/kubelet.conf` with the
+baseline cluster-wide configuration for all kubelets in the cluster. This configuration file
+points to the client certificates that allow the kubelet to communicate with the API server. This
+addresses the need to
+[propogate cluster-level configuration to each kubelet](#propagating-cluster-level-configuration-to-each-kubelet).
+
+To address the second pattern of
+[providing instance-specific configuration details](#providing-instance-specific-configuration-details),
+kubeadm writes an environment file to `/var/lib/kubelet/kubeadm-flags.env`, which contains a list of
+flags to pass to the kubelet when it starts. The flags are presented in the file like this:
 
-In order to address _Problem 2_, `kubeadm init` writes an environment file looking like this, in the path
-`/var/lib/kubelet/kubeadm-flags.env`:
 ```bash
 KUBELET_KUBEADM_ARGS="--flag1=value1 --flag2=value2 ..."
 ```
 
-kubeadm also writes to the file dynamic parameters like what cgroup driver that is used, or whether a non-docker
-`--cri-socket` is used. Then kubeadm attempts a `systemctl daemon-reload && systemctl restart kubelet` in order
-to make the kubelet pick up the latest dynamic flags which were previously written. After that the rest of the regular
-`kubeadm init` workflow continues.
+In addition to the flags used when starting the kubelet, the file also contains dynamic
+parameters such as the cgroup driver and whether to use a different CRI runtime socket
+(`--cri-socket`).
 
-On the other hand, `kubeadm join` uses the Bootstrap Token credential to download the `kubelet-config-1.X` ConfigMap
-and write it to `/var/lib/kubelet/config.yaml`.
+After marshalling these two files to disk, kubeadm attempts to run the following two
+commands, if you are using systemd to manage your processes:
 
-The dynamic environment file is generated in exactly the same way as `kubeadm init`.
+```bash
+systemctl daemon-reload && systemctl restart kubelet
+```
 
-After these files are written, `kubeadm` attempts a `systemctl daemon-reload && systemctl restart kubelet` in order
-to make the kubelet pick up the latest dynamic flags written.
+If the reload and restart are successful, the normal `kubeadm init` workflow continues.
 
-Then `kubeadm join` performs the discovery, which yields the `/etc/kubernetes/bootstrap-kubelet.conf` KubeConfig
-file consisting of a CA certificate and Bootstrap Token.
+### Workflow when using `kubeadm join`
 
-At that point the Bootstrap Token is used by the kubelet to perform the TLS Bootstrap and to get the unique credential which
-is stored in `/etc/kubernetes/kubelet.conf`. As of kubeadm v1.11, `kubeadm join` waits for the `/etc/kubernetes/kubelet.conf`
-file to appear on disk, which means that the kubelet has performed the TLS Bootstrap.
+When you run `kubeadm join`, kubeadm uses the Bootstrap Token credential to download the
```

Comment:
the bootstrap token is use to perform a TLS bootstrap which fetches the credential.
the original is more technically accurate.